src/libraries/System.Private.CoreLib/src/System/Threading/DeferredDisposableLifetime.cs

```diff
@@ -50,7 +50,7 @@ public bool AddRef()
                 int oldCount = Volatile.Read(ref _count);
 
                 // Have we been disposed?
-                ObjectDisposedException.ThrowIf(oldCount < 0, this);
+                ObjectDisposedException.ThrowIf(oldCount < 0, typeof(T));
```

Comment:
This assume that `typeof(T)` is "free". It is not the case here. This is shared generic code so `typeof(T)` is going to be generic dictionary lookup followed by helper call to compute the Type instance. The JIT is not smart enough to reorder this operation after the check of the condition.

It should be changed back to what it used to be and avoid using the throw helper.